### PR TITLE
Update SDK to preview 1 release

### PR DESCRIPTION
**Pull request by am11**
image

cc @ViktorHofer, @mmitche

### CODE REVIEW

File: src/Microsoft.DotNet.Arcade.Sdk/Microsoft.DotNet.Arcade.Sdk.csproj [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
     <Description>Common toolset for repositories</Description>
     <PackageTags>Arcade Build Repository Toolset MSBuild SDK</PackageTags>
+    <PackageProjectUrl>https://github.com/dotnet/arcade</PackageProjectUrl>
```
Comment (am11) [inline src/Microsoft.DotNet.Arcade.Sdk/Microsoft.DotNet.Arcade.Sdk.csproj:15]:
```suggestion
    <RepositoryUrl>https://github.com/dotnet/arcade</RepositoryUrl>
```

**Review by mmitche — APPROVED** [review]

**Review by ViktorHofer — APPROVED** [review]

File: Directory.Build.props [hunk 2]
```diff
@@ -13,6 +13,7 @@
     <IsShipping>false</IsShipping>
     <EnableNETAnalyzers>true</EnableNETAnalyzers>
     <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
+    <RepositoryUrl>https://github.com/dotnet/arcade</RepositoryUrl>
```
Comment (ViktorHofer) [inline Directory.Build.props:16]: @tmat I wonder what changed that made this necessary.

Comment (am11) [reply Directory.Build.props:16]: The error looked like this (for multiple projects):
> /__w/1/s/artifacts/sb/package-cache/microsoft.dotnet.arcade.sdk/9.0.0-beta.24106.2/tools/Workarounds.targets(79,5): error : PackageProjectUrl must be specified [/__w/1/s/artifacts/sb/src/src/Microsoft.DotNet.Arcade.Sdk/Microsoft.DotNet.Arcade.Sdk.csproj]

Comment (ViktorHofer) [reply Directory.Build.props:16]: I think https://github.com/dotnet/sourcelink/commit/064732082224f37fab77290f762be9b4a51f7dd9 caused this but am not 100% sure.

Comment (ViktorHofer) [reply Directory.Build.props:16]: OK it's actually a different change. See discussion in https://github.com/dotnet/runtime/pull/98476#issuecomment-1966396680.

**Pull request merged**